brage-import/src/main/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumer.java

```diff
@@ -91,15 +102,63 @@ private static Publication getOnlyElement(List<Publication> publications) {
                        fail.getException()));
     }
 
-    private boolean publicationWithCristinIdentifierAlreadyExists(Publication publication) {
+    private boolean shouldMergePublications(Publication publication) throws NotFoundException {
         var cristinIdentifier = getCristinIdentifier(publication);
         if (nonNull(cristinIdentifier)) {
-            publicationsByCristinIdentifier = getPublicationsByCristinIdentifier(cristinIdentifier);
-            return !publicationsByCristinIdentifier.isEmpty();
+            publicationsToMerge = getPublicationsByCristinIdentifier(cristinIdentifier);
+            return !publicationsToMerge.isEmpty();
+        }
+
+        return existingPublicationHasSameDoi(publication);
+    }
+
+    private boolean existingPublicationHasSameDoi(Publication publication)
+        throws NotFoundException {
+        var doi = publication.getEntityDescription().getReference().getDoi();
+
+        if (nonNull(doi)) {
+            var publications = searchForPublicationsByDoi(doi);
+            if (publications.isEmpty()) {
+                return false;
+            }
+
+            // Fetch first publication regardless of how many hits
+            var publicationByDoi = publications.get(0);
+
+            // The Search Index is not reliable enough, therefore we fetch the publication from the source
```

Comment:
Loose the comment and name your variable something like "mostUpdatedVersionOfPublication".

Comments are bad because we never know we refactor something, and they are no longer true.

Reply:
Or you can write a test where the search-index response is out of sync with the database, and you verify it is the database version you have updated.